src/MySqlConnector/MySqlClient/ConnectionPool.cs

```diff
@@ -91,15 +85,47 @@ private bool SessionIsHealthy(MySqlSession session)
 
 		public void Return(MySqlSession session)
 		{
+			// NOTE: deliberately calling 'async void' method so that first part runs synchronously, but remainder gets queued
```

Comment:
Is `async void` a recommended way of queuing background tasks?  Or would calling an `async Task` and not awaiting it have the same effect?  If it's all the same, I think the latter is better for consistency's sake.